plugins/ipam/commands/commands.go

```diff
@@ -100,17 +100,29 @@ func Del(args *skel.CmdArgs) error {
 }
 
 func del(ipManager ipstore.IPAllocator, ipamConf *config.IPAMConfig) error {
-	if ipamConf.IPV4Address.IP == nil {
-		return errors.New("del commands: ip address is required for deletion")
-	}
+	var releasedIP string
 
-	err := ipManager.Release(ipamConf.IPV4Address.IP.String())
-	if err != nil {
-		return err
+	if ipamConf.ID == "" && (ipamConf.IPV4Address.IP == nil || net.ParseIP(ipamConf.IPV4Address.IP.String()) == nil) {
```

Comment:
Can you move this block to `Del()` and refactor  config parsing part of `Del()` into a separate method?